comp/haagent/impl/haagent.go

```diff
@@ -11,23 +11,28 @@ import (
 
 	log "github.com/DataDog/datadog-agent/comp/core/log/def"
 	haagent "github.com/DataDog/datadog-agent/comp/haagent/def"
+	"github.com/DataDog/datadog-agent/comp/metadata/inventoryagent"
 	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
 	"github.com/DataDog/datadog-agent/pkg/util/hostname"
 	"go.uber.org/atomic"
 )
 
 type haAgentImpl struct {
 	log            log.Component
+	inventoryAgent inventoryagent.Component
 	haAgentConfigs *haAgentConfigs
 	state          *atomic.String
 }
 
-func newHaAgentImpl(log log.Component, haAgentConfigs *haAgentConfigs) *haAgentImpl {
-	return &haAgentImpl{
+func newHaAgentImpl(log log.Component, inventoryAgent inventoryagent.Component, haAgentConfigs *haAgentConfigs) *haAgentImpl {
+	haagentImpl := &haAgentImpl{
 		log:            log,
+		inventoryAgent: inventoryAgent,
 		haAgentConfigs: haAgentConfigs,
 		state:          atomic.NewString(string(haagent.Unknown)),
 	}
+	haagentImpl.setState(haagent.Unknown)
```

Comment:
I don't get the need for this since the state is already set to unknown in the line above ?